core/state/statedb.go

```diff
@@ -1373,64 +1367,91 @@ func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, er
 		if dst != nil {
-			al.AddAddress(*dst)
+			al.AddAddress(*dst, ALAllItems, false)
 			// If it's a create-tx, the destination will be added inside evm.create
 		}
 		for _, addr := range precompiles {
-			al.AddAddress(addr)
+			al.AddAddress(addr, 0, false)
 		}
 		for _, el := range list {
-			al.AddAddress(el.Address)
+			al.AddAddress(el.Address, 0, false)
 			for _, key := range el.StorageKeys {
-				al.AddSlot(el.Address, key)
+				al.AddSlot(el.Address, key, false)
 			}
 		}
 		if rules.IsShanghai { // EIP-3651: warm coinbase
-			al.AddAddress(coinbase)
+			al.AddAddress(coinbase, ALAllItems, false)
 		}
 	}
 	// Reset transient storage at the beginning of transaction execution
 	s.transientStorage = newTransientStorage()
 }
 
 // AddAddressToAccessList adds the given address to the access list
-func (s *StateDB) AddAddressToAccessList(addr common.Address) {
-	if s.accessList.AddAddress(addr) {
+func (s *StateDB) AddAddressToAccessList(addr common.Address, mode ALAccountItem, isWrite ALAccessMode) uint64 {
+	gas := s.accessList.AddAddress(addr, mode, isWrite)
+	if gas > 0 {
 		s.journal.append(accessListAddAccountChange{&addr})
 	}
+	return gas
 }
 
 // AddSlotToAccessList adds the given (address, slot)-tuple to the access list
-func (s *StateDB) AddSlotToAccessList(addr common.Address, slot common.Hash) {
-	addrMod, slotMod := s.accessList.AddSlot(addr, slot)
-	if addrMod {
-		// In practice, this should not happen, since there is no way to enter the
-		// scope of 'address' without having the 'address' become already added
-		// to the access list (via call-variant, create, etc).
-		// Better safe than sorry, though
-		s.journal.append(accessListAddAccountChange{&addr})
-	}
-	if slotMod {
+func (s *StateDB) AddSlotToAccessList(addr common.Address, slot common.Hash, isWrite ALAccessMode) uint64 {
+	gas := s.accessList.AddSlot(addr, slot, isWrite)
+	if gas > params.WarmStorageReadCostEIP2929 {
 		s.journal.append(accessListAddSlotChange{
 			address: &addr,
 			slot:    &slot,
 		})
 	}
+	return gas
+}
+
+func (s *StateDB) TouchTxOriginAndComputeGas(addr common.Address) uint64 {
+	return s.accessList.TouchTxOriginAndComputeGas(addr.Bytes())
+}
+
+func (s *StateDB) TouchTxExistingAndComputeGas(addr common.Address, sendsValue bool) uint64 {
+	return s.accessList.TouchTxExistingAndComputeGas(addr.Bytes(), sendsValue)
+}
+
+func (s *StateDB) TouchAndChargeContractCreateInit(addr common.Address, sendsValue bool) uint64 {
+	return s.accessList.TouchAndChargeContractCreateInit(addr.Bytes(), sendsValue)
+}
+
+func (s *StateDB) TouchAndChargeValueTransfer(from, to common.Address) uint64 {
+	return s.accessList.TouchAndChargeValueTransfer(from.Bytes(), to.Bytes())
+}
+
+func (s *StateDB) TouchAddressOnReadAndComputeGas(from common.Address, index uint64) uint64 {
+	treeIndex := uint256.NewInt((index + 128) / 256)
+	subIndex := byte((index + 128) % 256)
+	return s.accessList.TouchAddressOnReadAndComputeGas(from.Bytes(), *treeIndex, subIndex)
+}
```

Comment:
Can we rename this method and parameter name to: `TouchAddressCodeOnReadAndComputeGas(from common.Address, codeChunk uint64)` ?

This method is only used when touching chunks when executing opcodes, so the method name sounds confusing to me, and the parameter name is quite generic considering that here internally we do `+128` which only makes sense t for code.

Reply:
ah yes, I did it in the other PR that is against geth's master